src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/New-AzConnectedKubernetes.ps1

```diff
@@ -526,6 +544,26 @@ function New-AzConnectedKubernetes {
         } catch {
             throw "Unable to install helm chart at $ChartPath"
         }
+
+        if ($PSBoundParameters.ContainsKey('OidcIssuerProfileEnabled') -or $PSBoundParameters.ContainsKey('WorkloadIdentityEnabled') ) {
+            $ExistConnectedKubernetes = Get-AzConnectedKubernetes -ResourceGroupName $ResourceGroupName -ClusterName $ClusterName @CommonPSBoundParameters
+
+            Write-Host "Cluster configuration is in progress..."
+            $timeout = [datetime]::Now.AddMinutes(60)
+
+            while (($ExistConnectedKubernetes.ArcAgentProfileAgentState -ne "Succeeded") -and ($ExistConnectedKubernetes.ArcAgentProfileAgentState -ne "Failed") -and ([datetime]::Now -lt $timeout)) {
+                Start-Sleep -Seconds 30
+                $ExistConnectedKubernetes = Get-AzConnectedKubernetes -ResourceGroupName $ResourceGroupName -ClusterName $ClusterName @CommonPSBoundParameters
+            }
+
+            if ($ExistConnectedKubernetes.ArcAgentProfileAgentState -eq "Succeeded") {
+                Write-Host "Cluster configuration succeeded."
+            } elseif ($ExistConnectedKubernetes.ArcAgentProfileAgentState -eq "Failed") {
```

Comment:
If Failed or timeout, I think we should use write-error or as least write-warning instead of write-host.

Reply:
Thank you! Updated relevant lines as you suggested.